modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloader.java

```diff
@@ -177,18 +179,30 @@ void updateTaskState() {
     }
 
     //visible for testing
-    int indexChunks(String name, InputStream is, int chunk, String expectedMd5, long timestamp) throws IOException {
+    int indexChunks(String name, InputStream is, int startChunk, String expectedMd5, long timestamp) throws IOException {
+        int chunk = startChunk;
         MessageDigest md = MessageDigests.md5();
         for (byte[] buf = getChunk(is); buf.length != 0; buf = getChunk(is)) {
             md.update(buf);
-            client.prepareIndex(DATABASES_INDEX).setId(name + "_" + chunk + "_" + timestamp)
-                .setCreate(true)
-                .setSource(XContentType.SMILE, "name", name, "chunk", chunk, "data", buf)
-                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-                .setWaitForActiveShards(ActiveShardCount.ALL)
-                .get();
+            IndexRequest indexRequest = new IndexRequest(DATABASES_INDEX)
+                .id(name + "_" + chunk + "_" + timestamp)
+                .create(true)
+                .source(XContentType.SMILE, "name", name, "chunk", chunk, "data", buf)
+                .waitForActiveShards(ActiveShardCount.ALL);
+            client.index(indexRequest).actionGet();
             chunk++;
         }
+
+        if (chunk != startChunk) {
```

Comment:
I don't think we need this check really, chances that we have no chunks are marginal (and rather an error state), and even then no-op flush and refresh shouldn't cause much overhead.